### PR TITLE
Implement Mithril client WASM API

### PR DESCRIPTION
**Pull request by jpraynaud**
## Content
This PR includes:
- A new crate `mithril-client-wasm` that builds into WASM.
- Packaging tooling to release it as a `npm` package.
- An example website for using the Mithril client library in a browser.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [x] Update README file (if relevant)
  - [x] Update documentation website (if relevant)

## Issue(s)
Relates to #1336

### CODE REVIEW

File: mithril-client-wasm/tests/certificate_get_list.rs
Comment (dlachaumepalo) [inline mithril-client-wasm/tests/certificate_get_list.rs:1]: Do we need to keep this file?

File: mithril-client-wasm/src/client_wasm.rs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        let stake_distribution =
+            serde_wasm_bindgen::from_value(stake_distribution).map_err(|err| format!("{err:?}"))?;
+        let result = MessageBuilder::new()
+            .compute_mithril_stake_distribution_message(&stake_distribution)
+            .map_err(|err| format!("{err:?}"))?;
+
+        Ok(serde_wasm_bindgen::to_value(&result)?)
+    }
+
+    ///
```
Comment (dlachaumepalo) [inline mithril-client-wasm/src/client_wasm.rs:139]: We need to add a comment, just a suggestion, you'll probably have a better idea 😅.
```suggestion
    /// Call the client to validate the computed message
```

File: mithril-client-wasm/Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+CARGO = cargo
+
+all: test build
+
+build:
+	wasm-pack build --release --out-name index
+
+test:
+	#wasm-pack test --headless --firefox --chrome --node --release
```
Comment (dlachaumepalo) [inline mithril-client-wasm/Makefile:16]: Do we need to keep this comment?

File: examples/client-mithril-stake-distribution/Cargo.toml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 [package]
 name = "client-mithril-stake-distribution"
 description = "Mithril client stake distribution example"
-version = "0.1.0"
-authors = ["dev@iohk.io"]
+version = "0.1.1"
```
Comment (dlachaumepalo) [inline examples/client-mithril-stake-distribution/Cargo.toml:4]: Does this crate version need to be incremented? Even if the example code has not been modified.

File: docs/website/root/manual/developer-docs/nodes/mithril-client-library-wasm.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+```bash
+npm i @mithril-dev/mithril-client-wasm
+```
+
+## Using Mithril client library
+
+Below is a basic example of how to use most of the functions exposed by the Mithril client library:
+
+```js
+import { MithrilClient } from "@mithril-dev/mithril-client-wasm"
```
Comment (dlachaumepalo) [inline docs/website/root/manual/developer-docs/nodes/mithril-client-library-wasm.md:75]: Is this example enough for now? We also have the example in the www/ directory, which includes interaction with the DOM.

**Review by dlachaumepalo — APPROVED** [review]
LGTM 👍! Just few comments.

File: mithril-client-wasm/tests/certificate_get_list.rs
Comment (jpraynaud) [reply mithril-client-wasm/tests/certificate_get_list.rs:1]: Removed it, good catch!

File: mithril-client-wasm/src/client_wasm.rs (hunk 1 above)
Comment (jpraynaud) [reply mithril-client-wasm/src/client_wasm.rs:139]: Done :heavy_check_mark:

File: mithril-client-wasm/Makefile (hunk 2 above)
Comment (jpraynaud) [reply mithril-client-wasm/Makefile:16]: I removed it :+1:

File: docs/website/root/manual/developer-docs/nodes/mithril-client-library-wasm.md (hunk 4 above)
Comment (jpraynaud) [reply docs/website/root/manual/developer-docs/nodes/mithril-client-library-wasm.md:75]: Let's keep it simple :slightly_smiling_face:

**Review by ghubertpalo — APPROVED** [review]

**Pull request merged**